components/camel-soroush/pom.xml

```diff
@@ -0,0 +1,278 @@
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~
+  -->
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.apache.camel</groupId>
+        <artifactId>components</artifactId>
+        <version>3.0.0-SNAPSHOT</version>
+    </parent>
+
+
+    <artifactId>camel-soroush</artifactId>
+    <packaging>jar</packaging>
+    <name>Camel :: SoroushBot</name>
+
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <kotlin.version>1.3.21</kotlin.version>
+        <kotlin.compiler.incremental>true</kotlin.compiler.incremental>
+        <jersey-version>2.28</jersey-version>
+        <!--<jetty-version>9.4.15.v20190215</jetty-version>-->
+        <camel-version></camel-version>
+
+    </properties>
+
+
+    <!--
+        <dependencyManagement>
+            <dependencies>
+                &lt;!&ndash; Camel BOM &ndash;&gt;
+                <dependency>
+                    <groupId>org.apache.camel</groupId>
+                    <artifactId>camel-parent</artifactId>
+                    <version>2.23.1</version>
+                    <scope>import</scope>
+                    <type>pom</type>
+                </dependency>
+            </dependencies>
+        </dependencyManagement>
+    -->
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-support</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish.jersey.core</groupId>
+            <artifactId>jersey-client</artifactId>
+            <version>${jersey-version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-package-maven-plugin</artifactId>
+            <version>3.0.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish.jersey.media</groupId>
```

Comment:
Is Jersey the only possible solution here? Jersey is not OSGi friendly

Reply:
@oscerd 
No, but we build most of our logic around Jersey. We only use Jersey client in main project. Is it still hard to integrate it with Apache Karaf?

Reply:
From what I remember it's not really OSGi-friendly, but you can try to create the karaf feature for this component and check if it works with an integration test like what we have in tests/camel-itest-karaf

Reply:
@oscerd I add component to feature.xml and created and run a test it. Everything goes fine!